### PR TITLE
Fetch metadata for a single topic rather than all topics

### PR DESCRIPTION
**Pull request by mjtv**
Improves performance

### CODE REVIEW

**Comment by mjtv** [comment]
Note: This pull request is for performing a review only. We won't merge this into our master but rather will rebase our tivo-build off of this.

**Review by wushujames — APPROVED** [review]

**Review by manjeetvk — APPROVED** [review]
looks good

File: kafka/client.go [hunk 1]
```diff
@@ -472,18 +474,23 @@ func (client *Client) ReadTopic(name string, refreshMetadata bool) (Topic, error
...
+		if err == sarama.ErrUnknownTopicOrPartition {
+			err := TopicMissingError{msg: fmt.Sprintf("%s could not be found", name)}
+			return topic, err
+		}
 		if err != nil {
-			log.Printf("[ERROR] Error refreshing metadata %s", err)
+			log.Printf("[ERROR] Error refreshing topic '%s' metadata %s", name, err)
 			return topic, err
 		}
 		err = client.extractTopics()
```
Comment (llamahunter) [inline kafka/client.go:488]: this will clear knowledge on the client of other topics, correct?

Comment (mjtv) [reply kafka/client.go:488]: No. Data about the other topics is retained. Sarama has a cache (map) of topic data and only the topics passed to the RefreshMetadata method get updated in the cache.

**Review by llamahunter — APPROVED** [review]

**Pull request closed without merging**